tests/plugins/test_push_floating_tags.py

```diff
@@ -0,0 +1,497 @@
+            if 'secret' in registry_conf[docker_uri]:
+                reg_secret = registry_conf[docker_uri]['secret']
+
+            new_reg = {}
+            if reg_secret:
+                new_reg['auth'] = {'cfg_path': reg_secret}
+            else:
+                new_reg['auth'] = {'cfg_path': str(temp_dir)}
+            new_reg['url'] = 'https://' + docker_uri + '/' + reg_ver
+
+            registries_list.append(new_reg)
+
+        platform_descriptors_list = []
+        for platform in goarch:
+            new_plat = {
+                'platform': platform,
+                'architecture': goarch[platform],
+            }
+            platform_descriptors_list.append(new_plat)
+
+        workflow.plugin_workspace[ReactorConfigPlugin.key] = {}
+        workflow.plugin_workspace[ReactorConfigPlugin.key][WORKSPACE_CONF_KEY] =\
+            ReactorConfig({'version': 1,
+                           'registries': registries_list,
+                           'platform_descriptors': platform_descriptors_list})
+
+    runner = ExitPluginsRunner(tasker, workflow, plugins_conf)
+    results = runner.run()
+    plugin_result = results[PushFloatingTagsPlugin.key]
+
+    if expected_exception is None:
+        manifest_type, list_type = {
+            'v2': (
+                'application/vnd.docker.distribution.manifest.v2+json',
+                'application/vnd.docker.distribution.manifest.list.v2+json',
+            ),
+            'oci': (
+                'application/vnd.oci.image.manifest.v1+json',
+                'application/vnd.oci.image.index.v1+json',
+            ),
+        }[schema_version]
+
+        def verify_manifest_in_repository(registry, repo, manifest, platform, tag=None):
+            config = 'config-' + platform
+            assert registry.get_blob(repo, make_digest(config)) == config
+            layer = 'layer-' + platform
+            assert registry.get_blob(repo, make_digest(layer)) == layer
+            assert registry.get_manifest(repo, make_digest(manifest)) == manifest
+            if tag is not None:
+                assert registry.get_manifest(repo, tag) == manifest
+
+        source_builds = {}
+        source_manifests = {}
+
+        for platform in workers:
+            build = annotations['worker-builds'][platform]['digests'][0]
+            source_builds[platform] = build
+            source_registry = mocked_registries[build['registry']]
+            source_manifests[platform] = source_registry.get_manifest(build['repository'],
+                                                                      build['digest'])
+
+        for registry, _ in registry_conf.items():
+            target_registry = mocked_registries[registry]
+            for image in test_images:
+                name, tag = image.split(':')
+
+                if tag not in target_registry.get_repo(name)['tags']:
+                    continue
+
+                raw_manifest_list = to_text(target_registry.get_manifest(name, tag))
```

Comment:
and then we want to verify that this is the same exact thing as the manifest list available from the `test_images`.

Reply:
We could do something like this before the loop:
```
primary_tag = test_images[0].split(':', 1)[1]
primary_manifest_list = target_registry.get_manifest(name, primary_tag)
```

Then here, in the loop, check it's the same value:
```
assert target_registry.get_manfiest(name, tag) == primary_manifest_list
```

Reply:
Fixed